### ISSUE TITLE
Using with prettier API

### ISSUE THREAD

**Issue by shreyas-a** [prettier/prettier#4720]
```
prettier.format('php code', {
   plugins: ['prettier/plugin-php'],
});
```

This throws an error - No parser and no filepath given, using 'babylon' the parser now but this will throw an error in the future. Please specify a parser or a filepath so one can be inferred.

**Comment by ikatyang** [comment]
```js
prettier.format('php code', {
   filepath: "something.php", // or parser: "php"
   plugins: ['@prettier/plugin-php'],
});
```

**Comment by shreyas-a** [comment]
@ikatyang shows `Couldn't resolve parser "php"`

**Comment by ikatyang** [comment]
Can you share some environment info and steps to reproduce? (e.g. did you install the php plugin? global or local? etc.)

**Comment by shreyas-a** [comment]
Yeah, I've installed it locally.
```
"prettier": "^1.13.4",
"@prettier/plugin-php": "github:prettier/plugin-php",
```

**Comment by ikatyang** [comment]
I still cannot reproduce the issue, here's how I tried:

```sh
mkdir prettier-issue-4720
cd prettier-issue-4720
yarn init -y
yarn add prettier@1.13.4 github:prettier/plugin-php
echo '
const prettier = require("prettier");
const example = `
<?php
                    echo "Hello World!";
?>
something
`
console.log(prettier.format(example, {
  filepath: "something.php", // or parser: "php"
  plugins: ["@prettier/plugin-php"]
}));
' > test.js
node test.js
```

```php
<?php echo "Hello World!"; ?>
something
```

**Comment by shreyas-a** [comment]
When I pass `filepath: 'something.php'` is gives - `TypeError: path$8.extname is not a function`.

**Comment by ikatyang** [comment]
It seems you're using the `standalone.js`, which is [supposed to be used in the browser](https://prettier.io/blog/2018/05/27/1.13.0.html#prettier-works-in-the-browser). If you'd like to use it in the browser, you'll have to request this feature in [prettier/plugin-php](https://github.com/prettier/plugin-php).

**Comment by shreyas-a** [comment]
Thanks, as you mentioned I was using `standalone.js`, no errors or warnings as of now but the code is not getting formatted.

**Comment by duailibe** [comment]
@ikatyang I'm not sure what we should do.. the only use case we thought was passing the `parser`, but it wouldn't be hard to shim `path.extname()`

**Comment by ikatyang** [comment]
Closing as the `filepath` issue is fixed in #4721 and the browser support for php plugin should be tracked in [prettier/plugin-php](https://github.com/prettier/plugin-php).
